code/networking/src/validate_ip/ValidateIp.java

```diff
@@ -0,0 +1,35 @@
+import java.util.Arrays;
+import java.util.regex.Pattern;
+
+public class ValidateIp {
+    public static void main(String[] args) {
+        assert (isValid("172.16.254.1"));
+        assert (!isValid("172.16.254.01"));
+        assert (isValid("2001:0db8:85a3:0000:0000:8a2e:0370:7334"));
+        assert (isValid("2001:db8:85a3:0:0:8A2E:0370:7334"));
+        assert (!isValid("02001:0db8:85a3:0000:0000:8a2e:0370:7334"));
```

Comment:
conventionally, it's better to make a separate testing file.

Reply:
Agreed. This is hardly a unit test though, so creating a test file seemed unnecessary.
https://github.com/hvsalesforce/cosmos/blob/patch-1/code/networking/src/validate_ip/validate_ip.c Followed in the assertion footsteps of the C implementation.